pynestml/utils/ast_utils.py

```diff
@@ -1769,20 +1776,26 @@ def replace_inline_expressions_through_defining_expressions(cls, definitions: Se
         from pynestml.visitors.ast_symbol_table_visitor import ASTSymbolTableVisitor
 
         for m in inline_expressions:
-            source_position = m.get_source_position()
-            for target in definitions:
-                matcher = re.compile(cls._variable_matching_template.format(m.get_variable_name()))
-                target_definition = str(target.get_rhs())
-                target_definition = re.sub(matcher, "(" + str(m.get_expression()) + ")", target_definition)
-                target.rhs = ModelParser.parse_expression(target_definition)
-                target.update_scope(m.get_scope())
-                target.accept(ASTSymbolTableVisitor())
-
-                def log_set_source_position(node):
-                    if node.get_source_position().is_added_source_position():
-                        node.set_source_position(source_position)
-
-                target.accept(ASTHigherOrderVisitor(visit_funcs=log_set_source_position))
+            if "mechanism" not in [e.namespace for e in m.get_decorators()]:
```

Comment:
Could you move this filtering step to the place where the function is called? Then this function does not have to change.